src/builder/modules/Generator_Builder.res

```diff
@@ -437,6 +437,51 @@ let fillBackgroundColor = (model: Model.t, fillStyle: string) => {
   }
 }
 
+let drawLine = (
+  model: Model.t,
+  (x0, y0): position,
+  (x, y): position,
+  ~color: string,
+  ~width: float,
+  ~pattern: array<int>,
+  ~offset: int,
+) => {
+  let x1 = Belt.Int.toFloat(x0)
+  let y1 = Belt.Int.toFloat(y0)
+  let x2 = Belt.Int.toFloat(x)
+  let y2 = Belt.Int.toFloat(y)
+
+  let w = x2 -. x1
+  let h = y2 -. y1
+
+  let angle = w === 0.0 ? Js.Math._PI /. 2.0 : Js.Math.atan2(~y=h, ~x=w, ())
+  let ox = Js.Math.sin(angle) *. 0.5
+  let oy = Js.Math.cos(angle) *. 0.5
```

Comment:
Great work fixing this.

Reply:
Also, can you adding a comment with a link to a reference explaining the problem, so people reading this in the future can understand what the problem is, and how this bit of trigonometry fixes it, thanks.